### PR TITLE
Skew

### PR DESCRIPTION
**Pull request by asturur**
Skew functionality with no merge conflict.
Still have to put in all the tests.

### CODE REVIEW

**Comment by asturur** [comment]
cmon....
"Expected points but got points"  what he expect me to do?
Is point some kind of reserved word?
```
Expected points but got points, at src/util/misc.js :
   105 |    /**
   106 |     * Returns coordinates of points's bounding rectangle (left, top, width, height)
   107 |     * @param {Array} points, 4 points array
------------------------------^
   108 |     * @return {Object} Object with left, top, width, height properties
   109 |     */
```

**Comment by asturur** [comment]
@kangax ready to merge!

File: src/util/misc.js [hunk 1]
```diff
@@ -457,21 +482,44 @@
...
     },
 
     /**
+     * Decompone standard 2x2 matrix into transform componentes
+     * @static
+     * @memberOf fabric.util
+     * @param  {Array} a transformMatrix
+     * @return {Object} Components of transform
+     */
+    qrDecompone: function(a) {
```
Comment (kangax) [inline src/util/misc.js]: What's decompone?

Comment (asturur) [inline src/util/misc.js]: Is that function to normalize the control box
Mathematically is calle QR decomposition, and it's done on linear transform matrices.
Is not directly necessary for skewing, but is necessary to decompone the transformation, also is an handy tool for expert users that need to do something with transformMatrix.

We will use it later, it was already in the source.

Comment (kangax) [inline src/util/misc.js]: Hm, I don't think there's such word as decompone so let's change to decompose?

Comment (asturur) [inline src/util/misc.js]: mmm it is italian. i will put it in english.

**Pull request merged**